engine/src/flutter/lib/web_ui/lib/src/engine/lazy_path.dart

```diff
@@ -0,0 +1,718 @@
+      largeArc: largeArc,
+      clockwise: clockwise,
+    );
+  }
+}
+
+final class AddRectCommand implements PathCommand {
+  AddRectCommand(this.rect);
+
+  final ui.Rect rect;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addRect(rect);
+  }
+}
+
+final class AddOvalCommand implements PathCommand {
+  AddOvalCommand(this.oval);
+
+  final ui.Rect oval;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addOval(oval);
+  }
+}
+
+final class AddArcCommand implements PathCommand {
+  AddArcCommand(this.oval, this.startAngle, this.sweepAngle);
+
+  final ui.Rect oval;
+  final double startAngle;
+  final double sweepAngle;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addArc(oval, startAngle, sweepAngle);
+  }
+}
+
+final class AddPolygonCommand implements PathCommand {
+  AddPolygonCommand(this.points, this.close);
+
+  final List<ui.Offset> points;
+  final bool close;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addPolygon(points, close);
+  }
+}
+
+final class AddRRectCommand implements PathCommand {
+  AddRRectCommand(this.rrect);
+
+  final ui.RRect rrect;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addRRect(rrect);
+  }
+}
+
+final class AddRSuperellipseCommand implements PathCommand {
+  AddRSuperellipseCommand(this.rSuperellipse);
+
+  final ui.RSuperellipse rSuperellipse;
+
+  @override
+  void apply(DisposablePath path) {
+    path.addRSuperellipse(rSuperellipse);
+  }
+}
+
+final class AddPathCommand implements PathCommand {
+  AddPathCommand(this.path, this.offset, {this.matrix4});
+
+  final LazyPath path;
+  final ui.Offset offset;
+  final Float64List? matrix4;
+
+  @override
+  void apply(DisposablePath p) {
+    p.addPath(path.builtPath, offset, matrix4: matrix4);
+  }
+}
+
+final class ExtendWithPathCommand implements PathCommand {
+  ExtendWithPathCommand(this.path, this.offset, {this.matrix4});
+
+  final LazyPath path;
+  final ui.Offset offset;
+  final Float64List? matrix4;
+
+  @override
+  void apply(DisposablePath p) {
+    p.extendWithPath(path.builtPath, offset, matrix4: matrix4);
+  }
+}
+
+final class ClosePathCommand implements PathCommand {
+  @override
+  void apply(DisposablePath path) {
+    path.close();
+  }
+}
+
+abstract class DisposablePathConstructors {
+  DisposablePath createNew();
+  DisposablePath combinePaths(
+    ui.PathOperation operation,
+    DisposablePath path1,
+    DisposablePath path2,
+  );
+}
+
+class LazyPath implements ScenePath, Collectable {
+  factory LazyPath(DisposablePathConstructors constructors) =>
+      LazyPath._(constructors, () => constructors.createNew());
+  LazyPath._(this.constructors, this.initializer)
+    : fillType = ui.PathFillType.nonZero,
+      _commands = [];
+  LazyPath.fromLazyPath(LazyPath other)
+    : fillType = other.fillType,
+      constructors = other.constructors,
+      initializer = other.initializer,
+      _commands = List.from(other._commands);
```

Comment:
If there's a way to copy a `SkwasmPath` (and it's cheap), would it be beneficial to also copy `_cachedPath` to avoid re-applying all the commands later?